station_distance.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+require 'open-uri'
+
+def filescrapper(url)
+  csv_text = open(url)
+  CSV.parse(csv_text, headers: true, col_sep: ';')
+end
+
+def to_rad(angle)
+  angle.to_f * 0.017453292519943295
+end
+
+def distance_calc(coord)
+  a1 = Math.sin(coord[3] - coord[1])
+  a2 = Math.cos(coord[3]) * Math.cos(coord[1])
+  a3 = Math.sin(coord[2] - coord[0])
+  a = a1**2 + a2 * a3**2
+  b = Math.sqrt(a) / Math.sqrt(1 - a)
+  c = Math.atan(b)
+  (6371 * c).to_i
+end
+
+def station_user_selector(station_array)
+  station = []
+  p 'please select departure station id'
+  station << station_array[gets.chomp.to_i - 1]
+  p 'please select arrival station id'
+  station << station_array[gets.chomp.to_i - 1]
+end
+
+def station_select(station_list)
+  i = 0
+  station_array = []
+  station_list.each do |row|
```

Comment:
does `each_with_index` work here ? (i don't know, but it could save us 2 lines)

Reply:
it does B-)